### PR TITLE
DDCNL-4612| play28 upgrade wip

### PR DESCRIPTION
**Pull request by oliwiaKot**
- [ ] Unit tests passing
- [ ] Coverage reached
- [ ] Acceptance tests passing
- [ ] Reviewed

Need to run ATs once they are passing against master

### CODE REVIEW

File: app/controllers/YourIncomeCalculationController.scala [hunk 1]
```diff
@@ -108,7 +109,7 @@ class YourIncomeCalculationController @Inject()(
                 errorPagesHandler.internalServerError(
                   "Employment contains stub annual account data found meaning payment information can't be displayed")
               case (true, _) =>
-                Ok(views.html.print.historicIncomeCalculationView(historicIncomeCalculationViewModel, appConfig))
+                Ok(historicIncomeCalculationView(historicIncomeCalculationViewModel, appConfig))
```
Comment (struong) [inline app/controllers/YourIncomeCalculationController.scala:112]: This is a different view, its actually the print view coming in, we should probs rename the view class to represent this

File: test/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnectorSpec.scala [hunk 2]
```diff
@@ -44,7 +44,7 @@ class TaxFreeAmountComparisonConnectorSpec extends BaseSpec {
           "data" -> Json.obj(
             "previous" -> Json.arr(
               Json.obj(
-                "componentType" -> CarBenefit,
+                "componentType" -> Json.toJsFieldJsValueWrapper[TaxComponentType](CarBenefit),
```
Comment (struong) [inline test/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnectorSpec.scala:47]: Don't think `toJsFieldJsValueWrapper` is quite right here, this should be:

```suggestion
                "componentType" -> Json.toJson[TaxComponentType](CarBenefit),
```

File: test/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnectorSpec.scala [hunk 3]
```diff
@@ -54,7 +54,7 @@ class TaxFreeAmountComparisonConnectorSpec extends BaseSpec {
             ),
             "current" -> Json.arr(
               Json.obj(
-                "componentType" -> CarBenefit,
+                "componentType" -> Json.toJsFieldJsValueWrapper[TaxComponentType](CarBenefit),
```
Comment (struong) [inline test/uk/gov/hmrc/tai/connectors/TaxFreeAmountComparisonConnectorSpec.scala:57]:
```suggestion
                "componentType" -> Json.toJson[TaxComponentType](CarBenefit),
```

File: test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala [hunk 4]
```diff
@@ -33,7 +33,7 @@ class TaxFreeAmountComparisonSpec extends PlaySpec {
         val json = Json.obj(
           "previous" -> Json.arr(
             Json.obj(
-              "componentType" -> CarBenefit,
+              "componentType" -> Json.toJsFieldJsValueWrapper[TaxComponentType](CarBenefit),
```
Comment (struong) [inline test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala:36]:
```suggestion
              "componentType" -> Json.toJson[TaxComponentType](CarBenefit),
```

File: test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala [hunk 5]
```diff
@@ -43,7 +43,7 @@ class TaxFreeAmountComparisonSpec extends PlaySpec {
           ),
           "current" -> Json.arr(
             Json.obj(
-              "componentType" -> Mileage,
+              "componentType" -> Json.toJsFieldJsValueWrapper[TaxComponentType](Mileage),
```
Comment (struong) [inline test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala:46]:
```suggestion
              "componentType" -> Json.toJson[TaxComponentType](Mileage),
```

File: app/controllers/YourIncomeCalculationController.scala (hunk 1 above)
Comment (Christian-QA) [reply app/controllers/YourIncomeCalculationController.scala:112]: Done

File: test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala (hunk 5 above)
Comment (Christian-QA) [reply test/uk/gov/hmrc/tai/model/domain/TaxFreeAmountComparisonSpec.scala:46]: Done

**Review by struong — APPROVED** [review]

**Pull request closed without merging**